File: leetcode/medium/2466. Count Ways To Build Good Strings/main.go
package main

const MOD = 1e9 + 7

func countGoodStrings(low int, high int, zero int, one int) int {
	dp := make(map[int]int)

	var dfs func(length int) int
	dfs = func(length int) int {
		if length > high {
			return 0
		}

		if val, ok := dp[length]; ok {
			return val
		}

		var res int
		if length >= low {
			res = 1
		}

		res += dfs(length+zero) + dfs(length+one)

		dp[length] = res % MOD

		return res % MOD
	}

	return dfs(0)
}

File: leetcode/medium/2466. Count Ways To Build Good Strings/main.ts
const MOD = 1e9 + 7;

const countGoodStrings = (low: number, high: number, zero: number, one: number): number => {
  let dp: { [key: number]: number } = {};

  const dfs = (length: number): number => {
    if (length > high) {
      return 0
    }

    if (length in dp) {
      return dp[length]
    }

    let count = (length >= low) ? 1 : 0;
    count += dfs(length + zero);
    count %= MOD;
    count += dfs(length + one);
    count %= MOD;

    dp[length] = count;

    return count;
  }

  return dfs(0);
};
